packages/jellyfish-api-core/__tests__/category/loan/listAuctons.test.ts

```diff
@@ -0,0 +1,521 @@
+      }
+      // including_start = false
+      {
+        const page = await testing.rpc.loan.listAuctions(
+          { start: { height: auction4.liquidationHeight } }
+        )
+        expect(page.length).toStrictEqual(0)
+      }
+    })
+
+    it('should listAuctions with vaultId', async () => {
+      // including_start = true
+      {
+        const page = await testing.rpc.loan.listAuctions(
+          { start: { vaultId: auction1.vaultId }, including_start: true }
+        )
+        expect(page.length).toStrictEqual(4) // Unable to filter by vaultId only. Need to filter both vaultId and height together.
+      }
+      // including_start = false
+      {
+        const page = await testing.rpc.loan.listAuctions(
+          { start: { vaultId: auction1.vaultId } }
+        )
+        expect(page.length).toStrictEqual(3) // Unable to filter by vaultId only. Need to filter both vaultId and height together.
+      }
+    })
+
+    it('should listAuctions with vaultId and height', async () => {
+      // List for liquidation height of second vault and vaultId of second vault
+      // including_start = true
+      {
+        const page = await testing.rpc.loan.listAuctions(
+          { start: { vaultId: auction2.vaultId, height: auction2.liquidationHeight }, including_start: true }
+        )
+        expect(page.length).toStrictEqual(3)
+        expect(page[0].vaultId).toStrictEqual(auction2.vaultId)
+        expect(page[1].vaultId).toStrictEqual(auction3.vaultId)
+        expect(page[2].vaultId).toStrictEqual(auction4.vaultId)
+      }
+      // including_start = false
+      {
+        const page = await testing.rpc.loan.listAuctions(
+          { start: { vaultId: auction2.vaultId, height: auction2.liquidationHeight } }
+        )
+        expect(page.length).toStrictEqual(2)
+        expect(page[0].vaultId).toStrictEqual(auction3.vaultId)
+        expect(page[1].vaultId).toStrictEqual(auction4.vaultId)
+      }
+
+      // List for liquidation height of first vault and vaultId of second vault
+      // including_start = true
+      {
+        const page = await testing.rpc.loan.listAuctions(
+          { start: { vaultId: auction1.vaultId, height: auction2.liquidationHeight }, including_start: true }
+        )
+        expect(page.length).toStrictEqual(3)
+        expect(page[0].vaultId).toStrictEqual(auction2.vaultId)
+        expect(page[1].vaultId).toStrictEqual(auction3.vaultId)
+        expect(page[2].vaultId).toStrictEqual(auction4.vaultId)
+      }
+      // including_start = false
+      {
+        const page = await testing.rpc.loan.listAuctions(
+          { start: { vaultId: auction1.vaultId, height: auction2.liquidationHeight } }
+        )
+        expect(page.length).toStrictEqual(2)
+        expect(page[0].vaultId).toStrictEqual(auction3.vaultId)
+        expect(page[1].vaultId).toStrictEqual(auction4.vaultId)
+      }
+
+      // List for liquidation height of third vault and vaultId of second vault
```

Comment:
This is not recommended too.
Added test unit here and show that the result is unpredictable.